src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -49,6 +81,40 @@ def get_osd_map(self):
             osds[str(osd['id'])] = osd
         return osds
 
+    def parse_osd_devices(self, osd_metadata):
+        result = {}
+        if osd_metadata:
+            devs = osd_metadata.get('device_ids', '')
+            if devs:
+                result = {}
+                for osd_device_id in devs.split(","):
+                    if str(osd_device_id).split('=') >= 2:
+                        dev_name = osd_device_id.split('=')[0]
+                        dev_id = osd_device_id.split('=')[1]
+                    else:
+                        dev_id = osd_device_id
+                    result[dev_name] = {'dev_id': dev_id}
+        return result
+
+    def get_device_life_expectancy_weeks(self, device_id):
+        life_expectancy_weeks = None
+        try:
+            dev_info = CephService.send_command('mon', prefix='device info', devid=device_id)
```

Comment:
instead, self.get('device ' + devid). that won't block (we have the info in the mgr's memory)

Reply:
OK, Thanks.

Reply:
@liewegas 
It seems hard to pass the test item "run-tox-mgr-dashboard" if the dashboard queried device info.
106/154 Test   #6: run-tox-mgr-dashboard ...................***Failed   82.39 sec

Reply:
It's probably the use of send_command() here, since that command is actually implemented by a mgr module.  Switching to self.get('device ' + ..) should fix it.

Reply:
I already use self.get('device ' + ..) and I try he self.get('devices'). It still get make check fail.

Reply:
@hsiang41 which tests are failing? Do you have the make check output?

Reply:
@sebastian-philipp Let me trigger make check again.